tests/store/advocates.spec.ts

```diff
@@ -0,0 +1,153 @@
+  name: 'Max Mustermann',
+  region: 'Europe'
+}
+
+/**
+ * ACTIONS
+ * -----------------------------------------------------------------------------
+ */
+
+describe('fetchAdvocates', () => {
+  it('commits "setAdvocates" with a list of fetched elements', async () => {
+    const commit = jest.fn()
+
+    const fetchAdvocates = advocatesModule.actions.fetchAdvocates as Function
+    await fetchAdvocates({ commit })
+
+    expect(commit).toHaveBeenCalledWith(
+      'setAdvocates',
+      expect.arrayContaining([expect.anything()])
+    )
+  })
+})
+
+/**
+ * MUTATIONS
+ * -----------------------------------------------------------------------------
+ */
+
+describe('setAdvocates', () => {
+  const state = {
+    advocates: [],
+    regionFilters: []
+  }
+
+  it('sets the list of advocates with one advocate', () => {
+    advocatesModule.mutations.setAdvocates(state, [advocate1])
+    expect(state).toEqual({
+      advocates: [advocate1],
+      regionFilters: []
+    })
+  })
+
+  it('sets the list of advocates with one different advocate', () => {
+    advocatesModule.mutations.setAdvocates(state, [advocate2])
+    expect(state).toEqual({
+      advocates: [advocate2],
+      regionFilters: []
+    })
+  })
+
+  it('sets the list of advocates with multiple advocates', () => {
+    advocatesModule.mutations.setAdvocates(state, [advocate1, advocate2])
+    expect(state).toEqual({
+      advocates: [advocate1, advocate2],
+      regionFilters: []
+    })
+  })
+
+  it('unsets the empty list of advocates', () => {
+    advocatesModule.mutations.setAdvocates(state, [])
+    expect(state).toEqual({
+      advocates: [],
+      regionFilters: []
+    })
+  })
+})
+
+describe('setRegionFilters', () => {
+  const state = {
+    advocates: [],
+    regionFilters: []
+  }
+  const regionFilter1 = 'South America'
+  const regionFilter2 = 'Europe'
+
+  it('sets the region filters with one filter', () => {
+    advocatesModule.mutations.setRegionFilters(state, [regionFilter1])
+    expect(state).toEqual({
+      advocates: [],
+      regionFilters: [regionFilter1]
+    })
+  })
+
+  it('sets the region filters with one different filter', () => {
+    advocatesModule.mutations.setRegionFilters(state, [regionFilter2])
+    expect(state).toEqual({
+      advocates: [],
+      regionFilters: [regionFilter2]
+    })
+  })
+
+  it('sets the region filters with multiple filters', () => {
+    advocatesModule.mutations.setRegionFilters(state, [regionFilter1, regionFilter2])
+    expect(state).toEqual({
+      advocates: [],
+      regionFilters: [regionFilter1, regionFilter2]
+    })
+  })
+
+  it('unsets the region filters', () => {
+    advocatesModule.mutations.setRegionFilters(state, [])
+    expect(state).toEqual({
+      advocates: [],
+      regionFilters: []
+    })
+  })
+})
+
+/**
+ * GETTERS
+ * -----------------------------------------------------------------------------
+ */
+
+describe('filteredAdvocates', () => {
```

Comment:
You are using state variables directly. You should access the state through getters, mutations, or actions to set the variables.
If the internal behavior is modified but the external keep the same, you will need to change these tests. A test should check external behavior and should be changed only if external behavior changes.

Reply:
I'm going to create Vuex stores before each test to test the functionality of mutations and getters over the Vuex API.